### PR TITLE
BPMN subprocess

### LINKED ISSUE #1118: I can run a workflow with an embedded subprocess
* activity instance records for the subprocess are created
* I can use io mappings on the subprocess
* Does initially not support cancellation of instances with active subprocesses; this is part of #1119

### PR DESCRIPTION
**Pull request by ThorbenLindhauer**
- no cancellation => #1119
- no payload update => #1131
- index changes maybe do not require an in-depth review since #1119 will replace all the indexes

closes #1118

### CODE REVIEW

File: broker-core/src/test/java/io/zeebe/broker/workflow/subprocess/EmbeddedSubProcessTest.java [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+    // when
+    testClient.createWorkflowInstance(PROCESS_ID, payload);
+
+    // then
+    final SubscribedRecord jobCreatedEvent = testClient.receiveFirstJobEvent(JobIntent.CREATED);
+    assertThat(jobCreatedEvent.value()).containsEntry("payload", expectedMappedPayload);
+  }
+
+  @Test
+  public void shouldApplyOutputMappings() {
```
Comment (Zelldon) [inline broker-core/src/test/java/io/zeebe/broker/workflow/subprocess/EmbeddedSubProcessTest.java]: Can we have also an test for both mapping types?

**Review by ChrisKujawa — APPROVED** [review]

**Comment by menski** [comment]
bors retry

**Pull request merged**